src/twisted/newsfragments/12068.bugfix

```diff
@@ -0,0 +1,2 @@
+Fixed invalid handling of ``env=None`` in ``reactor.spawnProcess`` on Posix
+when os.posix_spawnp() is being used.
```

Comment:
```suggestion
twisted.internet.process.Process, used by ``reactor.spawnProcess``, now copies the parent environment when the `env=None` argument is passed on Posix systems.
```

Is there a case of a supported OS when os.posix_spawnp is not used on Posix (Linux and macOS) ?

I guess that the fork code can be removed / deprecated.
Are there still posix OSes without `posix_spawnp` ?

Reply:
Unfortunately posix_spawnp does not support changing the current directory, uid and gid. Setting current directory has been accepted to POSIX (https://www.austingroupbugs.net/view.php?id=1208), but for uid and gid we'll need to use fork+execvpe for a long time it seems.

Reply:
Many thanks for the info.
Just a question.

Do you think it is worth updating the current documentation to explain that posix_spawnp is used when possible, and if you use path and uid/gid the good old fork + execvpe is used ?

Reply:
> Do you think it is worth updating the current documentation to explain that posix_spawnp is used when possible, and if you use path and uid/gid the good old fork + execvpe is used ?

If so, not in this PR :)

Reply:
> Do you think it is worth updating the current documentation to explain that posix_spawnp is used when possible, and if you use path and uid/gid the good old fork + execvpe is used ?

I think this is useful addition. Usage of os.fork is user visible because new python throws a warning, so end users may want to adjust their code to not use it (e.g. invoke spawnProcess differently).

I've opened https://github.com/twisted/twisted/pull/12072.

Reply:
@adiroiban I think it's important to mention that only posix_spawnp code path is being affected in the release note. I've used your wording as baseline and adjusted it to include posix_spawnp note.